whois-endtoend/src/test/java/net/ripe/db/whois/query/InverseQueryTestIntegration.java

```diff
@@ -266,7 +274,219 @@ public void inverse_email_from_untrusted_then_fail() {
+                "descr:       Owner Maintainer\n" +
+                "admin-c:     PP1-TEST\n" +
+                "upd-to:      noreply@ripe.net\n" +
+                "auth:        MD5-PW $1$d9fKeTr2$Si7YudNf4rUGmR71n/cqk/ #test\n" +
+                "auth:        SSO person@net.net\n" +
+                "auth:        PGPKEY-A8D16B70\n" +
+                "mnt-by:      OWNER-MNT\n" +
+                "source:      TEST"));
+
+        final String rpslObject = "" +
+                "person:         Pauleth Palthen\n" +
+                "address:        Singel 258\n" +
+                "phone:          +31-1234567890\n" +
+                "remarks:        A, B, Ѣ, H, O, Ꙋ\n" +
+                "mnt-by:         OWNER1-MNT\n" +
+                "nic-hdl:        PP2-TEST\n" +
+                "source:         TEST";
+
+        /*
+         * CP1251: A(41), B(42), Ѣ -> ?(3F), H(48), O(4F), Ꙋ -> ?(3F)
+         * */
+
+        databaseHelper.addObject(rpslObject);
+
+        final Pattern pattern = Pattern.compile("remarks:\s+(.*?)\s*\n");
+        final String cyrillicExpectedResult = getCharsetValuesInHex("A, B, Ѣ, H, O, Ꙋ", Charset.forName("CP1251"));
+
+        final String cyrillicEncodingResponse = query("-Z CP1251 -Bi mnt-by OWNER1-MNT", Charset.forName("CP1251"));
+        Matcher matcher = pattern.matcher(cyrillicEncodingResponse);
+        assertThat(matcher.find(), is(true));
+        assertThat(cyrillicExpectedResult, is(getCharsetValuesInHex(matcher.group(1), Charset.forName("CP1251"))));
+        assertThat("A, B, ?, H, O, ?", is(matcher.group(1)));
+    }
+
+    @Test
+    public void inverse_email_non_recognised_charset() {
+        databaseHelper.addObject(RpslObject.parse("" +
+                "mntner:      OWNER1-MNT\n" +
+                "descr:       Owner Maintainer\n" +
+                "admin-c:     PP1-TEST\n" +
+                "upd-to:      noreply@ripe.net\n" +
+                "auth:        MD5-PW $1$d9fKeTr2$Si7YudNf4rUGmR71n/cqk/ #test\n" +
+                "auth:        SSO person@net.net\n" +
+                "auth:        PGPKEY-A8D16B70\n" +
+                "mnt-by:      OWNER-MNT\n" +
+                "source:      TEST"));
+
+        final String rpslObject = "" +
+                "person:         Pauleth Palthen\n" +
+                "address:        Singel 258\n" +
+                "phone:          +31-1234567890\n" +
+                "remarks:        é, Ú, ß\n" +
+                "mnt-by:         OWNER1-MNT\n" +
+                "nic-hdl:        PP2-TEST\n" +
+                "remarks:        remark\n" +
+                "source:         TEST";
+        
+        databaseHelper.addObject(rpslObject);
+
+        assertThat(query("-Z nonExistCharset -Bi mnt-by OWNER1-MNT"), containsString("%ERROR:100: internal software error"));
+    }
+
+    private String getCharsetValuesInHex(final String values, final Charset charset){
+        byte[] charsetByte = String.valueOf(values).getBytes(charset);
```

Comment:
`String.valueOf(<String>)` is unnecessary. Remove method variable and just do `for (byte b : values.getBytes(charset)`

Reply:
As discussed, I suggest removing `getCharsetValuesInHex` altogether and just compare the `String` responses

Reply:
yes, done